acme/acme/client.py

```diff
@@ -527,21 +598,29 @@ def __del__(self):
         except Exception:  # pylint: disable=broad-except
             pass
 
-    def _wrap_in_jws(self, obj, nonce):
+    def _wrap_in_jws(self, obj, nonce, url, acme_version):
         """Wrap `JSONDeSerializable` object in JWS.
 
         .. todo:: Implement ``acmePath``.
 
         :param .JSONDeSerializable obj:
+        :param str url: The URL to which this object will be POSTed
         :param bytes nonce:
         :rtype: `.JWS`
 
         """
         jobj = obj.json_dumps(indent=2).encode()
         logger.debug('JWS payload:\n%s', jobj)
-        return jws.JWS.sign(
-            payload=jobj, key=self.key, alg=self.alg,
-            nonce=nonce).json_dumps(indent=2)
+        kwargs = {
+            "alg": self.alg,
+            "nonce": nonce
+        }
+        if acme_version == 2:
+            kwargs["url"] = url
+            kwargs["kid"] = self.account["uri"]
```

Comment:
Hold up, just noticed -- `messages.Registration` objects don't have a `uri`. What is this `uri`, and can we get it somewhere else?

Reply:
I think `account` needs to be a `messages.RegistrationResource` object not a `messages.Registration` object. Luckily, this is what the `register` method already returns, so I think we just need to update the documentation for `ClientNetwork`.